### DISCUSSION TITLE
macfuse unclear licensing/question about decleration of closed source casks

### DISCUSSION THREAD

**Discussion by JacksonChen666** [Homebrew/discussions#3673] [Casks] [answered]
so i'm just trying to solve `aft-mtp-mount` not finding the `fuse` executable. boy oh boy am i not getting the time back for finding out that macfuse is closed source (only through extensive research), macfuse not have a `fuse` binary (at least i couldn't find it), libfuse only works on linux, and writing this discussion.

### for main point, see proposal and questions

# events
- i need to transfer a lot of files to my android phone
  - i would do it in linux (on the same mac) but i would like to use my computer normally ([linux on apple silicon](https://asahilinux.org/) isn't practical yet)
  - using the gui application for `android-file-transfer-linux` (on macos) is not feasible since it is unstable enough to my disliking
- i use `aft-mtp-mount`
- `fuse` executable not found
- `brew info osxfuse` and it says in the notes to install macfuse instead
- `brew uninstall osxfuse` `brew install macfuse` (fuse seems to be working)
- i find that some formulas were disabled "because it requires closed-source macFUSE!"
- i go into a tangent and research for if macfuse is open source or not (spoiler: it is not after a certain version, see links section for more)
- i think to myself "hey, then why does the cask not have a note saying that it is closed source? it probably should since i expected it to be open source"
- i'm writing this discussion

# proposal
add a note/caveat saying that macfuse/osxfuse is no longer open source.

(ok yeah i could do that above by myself, but i have questions below)

# questions
- why is a closed source notice not added for casks? (or macFUSE specifically)
- should homebrew users expect to potentially install & use closed source software without their explicit knowledge that the software is closed source? (`brew install` accepts a formula or a cask, homebrew formulas (i guess) need to be open source but not casks)
- how can homebrew users prevent the installation of closed source software? or how can users know that they are getting closed source software?
  - if it cannot be achieved, is the user able to configure homebrew to prevent installation of casks unless explicitly providing a `--cask` argument for installation? (`brew install` `--formula` option exists, but it's not default)
- could i select multiple answers in github discussions?

# links
- https://github.com/Homebrew/homebrew-core/pull/74812
- https://github.com/orgs/Homebrew/discussions/1760#discussioncomment-945235
- https://github.com/osxfuse/osxfuse/issues/616#issuecomment-726427625
- https://github.com/Homebrew/homebrew-core/pull/64491
- https://github.com/Homebrew/homebrew-cask/issues?q=macfuse+license
- https://github.com/orgs/Homebrew/discussions/1497#discussioncomment-760897

**Comment by SMillerDev** [comment, marked as answer]
> why is a closed source notice not added for casks? (or macFUSE specifically)

Because they exist to facilitate install of tools that are only available as binaries, so that would be almost all of them.

> should homebrew users expect to potentially install & use closed source software without their explicit knowledge that the software is closed source? (brew install accepts a formula or a cask, homebrew formulas (i guess) need to be open source but not casks)

Only homebrew/core software needs to be open source. If you care about the licenses there's an environment variable to ensure one is set.

> how can homebrew users prevent the installation of closed source software? or how can users know that they are getting closed source software?

I guess I answered this above.
> if it cannot be achieved, is the user able to configure homebrew to prevent installation of casks unless explicitly providing a --cask argument for installation? (brew install --formula option exists, but it's not default)

You can untap `homebrew/cask`
> could i select multiple answers in github discussions?

Nope

**Reply by JacksonChen666** [reply]
> If you care about the licenses there's an environment variable to ensure one is set.

i was unable to find an environment variable specifically for not allowing installation unless a license is set.
i did find `HOMEBREW_FORBIDDEN_LICENSES`, but it's not obvious on what value it should be set for disallowing no licensing declaration.

**Reply by SMillerDev** [reply]
I thought there was an allowlist too. My mistake.
